src/pke/lib/encoding/ckkspackedencoding.cpp

```diff
@@ -494,71 +495,76 @@ bool CKKSPackedEncoding::Decode(size_t noiseScaleDeg, double scalingFactor, enum
 
     double logstd = std::log2(stddev);
 
-    // if stddev < sqrt{N}/8 (minimum approximation error that can be achieved)
-    if (stddev < 0.125 * std::sqrt(GetElementRingDimension())) {
-        stddev = 0.125 * std::sqrt(GetElementRingDimension());
+    if (executionMode == EXEC_NOISE_ESTIMATION) {
+        m_logError = logstd;
```

Comment:
I don't think this is right. The current error is in the coefficient (integer) representation, not in the encoded (real/complex) space. See how m_logError is set below.